velox/docs/functions/spark/misc_function.rst

```diff
@@ -0,0 +1,7 @@
+====================================
+Miscellaneous Functions
+====================================
+.. spark:function:: spark_partition_id() -> integer
+
+    Returns the current partition id.
+    This is non deterministic because it depends on data partitioning and task scheduling.
```

Comment:
This description is not quite accurate, is it?

I feel that it requires more details. See https://facebookincubator.github.io/velox/functions/spark/math.html#id5

It would be good to clarify that this function is marked as deterministic and Velox constant folds it. Would you also double check that constant folding happens indeed?

Reply:
Confirmed that constant folding happens. I added more details to this description, thanks.